text/0256-react-cdk-add-jsx-tsx-support.md

```diff
@@ -0,0 +1,182 @@
+---
+feature name: react-cdk-add-jsx-tsx-support
+start date: 2020-09-27
+rfc pr:
+related issue: https://github.com/aws/aws-cdk-rfcs/issues/256
+---
+
+# Summary
+
+Support the expression of cloud infrastructure resources via React JSX / TSX. Cloud solutions
+are composed of many services. Being able to efficiently and naturally express the composition
+is a developer ergonomics/DX concern.
+
+# README
+
+Create CDK applications using React.
+
+## Hello ReactCDK
+
+Example ReactCDK application
+
+```jsx
+const { ReactCDK, App, Stack } = require('@aws-cdk/react')
+const Bucket = require('@aws-cdk/react/s3')
+const { CloudFrontDistribution, Origins, DefaultOrigin } = require('@aws-cdk/react/cloudfront')
+const { Api, Resource, Integration } = require('@aws-cdk/react/apigateway')
+const Lambda = require('@aws-cdk/react/lambda')
+
+const EchoLambda = (
+  <Lambda>
+  {
+    async ({event, context}) => ({
+      "statusCode": 200,
+      "headers": {
+            "Content-Type": "application/json"
+        }
+      "body": JSON.stringify(event)
+    })
```

Comment:
In my opinion the code for a lambda should still be a separated file, and be passed to `<Lambda />` as a prop as a path to the code

e.g.
```tsx
<Lambda handler="src/handler" />
```